pages/calendar.js

```diff
@@ -0,0 +1,99 @@
+import { Calendar, momentLocalizer, Views } from "react-big-calendar";
+import moment from "moment";
+import LayoutStyle from "../components/LayoutStyle";
+import back from "../public/images/back.svg";
+import { useRouter } from "next/router";
+import Image from "next/image";
+import "react-big-calendar/lib/css/react-big-calendar.css";
+import styled from "styled-components";
+import { useMemo } from "react";
+import { getSession } from "next-auth/react";
+import Task from "../models/Task";
+import dbConnect from "../lib/dbConnect";
+
+export const getServerSideProps = async (context) => {
+  const session = await getSession(context);
+  if (!session) {
+    return {
+      redirect: {
+        destination: "/onboarding",
+      },
+    };
+  }
+
+  await dbConnect();
+
+  const result = await Task.find({ email: session.user.email });
+
+  const tasks = result.map((item) => {
+    const startDate = item.start;
+    const endDate = new Date(item.start.getTime() + item.totalTime * 1000);
+
+    return {
+      id: item.id,
+      title: item.name,
+      start: startDate.toJSON(),
+      end: endDate.toJSON(),
+    };
+  });
+  return { props: { tasks: tasks } };
+};
+
+export default function MyCalendar({ tasks }) {
+  const router = useRouter();
+  const localizer = momentLocalizer(moment);
+  const { defaultDate, views } = useMemo(
+    () => ({
+      defaultDate: new Date(),
+      views: {
+        day: true,
+      },
+    }),
+    []
+  );
+
+  const events = tasks.map((item) => ({
+    id: item.id,
+    title: item.title,
+    start: new Date(item.start),
+    end: new Date(item.end),
```

Comment:
you could spread the item here first and then overwrite start and end with the new Date versions of the values:
```js
{
...item,
start: new Date(...),
end: new Date (...)
}
```

Reply:
@F-Kirchhoff good suggestion! already improved 👍